src/librustc_passes/ast_validation.rs

```diff
@@ -237,16 +234,16 @@ impl<'a> Visitor<'a> for AstValidator<'a> {
                 if is_auto == IsAuto::Yes {
                     // Auto traits cannot have generics, super traits nor contain items.
                     if !generics.ty_params.is_empty() {
-                        self.err_handler().span_err(item.span,
-                                                    "auto traits cannot have generics");
+                        struct_span_err!(self.session, item.span, E0567,
+                                        "Auto traits cannot have type parameters").emit();
```

Comment:
"Auto" -> "auto", error messages are not normally capitalized.
Also "type parameters" -> "generic parameters" and the condition a few lines above is incorrect, it should use `!generics.is_parameterized()`, because lifetime parameters are disallowed as well.